Code/Editor/Platform/Linux/Editor/Core/QtEditorApplication_linux.h

```diff
@@ -6,19 +6,35 @@
  *
  */
 
+#if !defined(Q_MOC_RUN)
 #include <Editor/Core/QtEditorApplication.h>
+#include <AzToolsFramework/Entity/EditorEntityContextBus.h>
+#endif
+
+using xcb_connection_t = struct xcb_connection_t;
 
 namespace Editor
 {
-    class EditorQtApplicationXcb : public EditorQtApplication
+    class EditorQtApplicationXcb
+        : public EditorQtApplication
+        , public AzToolsFramework::EditorEntityContextNotificationBus::Handler
     {
         Q_OBJECT
     public:
         EditorQtApplicationXcb(int& argc, char** argv)
             : EditorQtApplication(argc, argv)
         {
+            // Connect bus to listen for OnStart/StopPlayInEditor events
+            AzToolsFramework::EditorEntityContextNotificationBus::Handler::BusConnect();
         }
 
```

Comment:
Should add a destructor to disconnect from the AzToolsFramework::EditorEntityContextNotificationBus as well

Reply:
`EditorEntityContextNotification` (the `EBusTraits` class used by that bus) does not override the `MutexType`, so it is using the default one set in `EBusTraits`, which is `NullMutex`. Explicitly disconnecting from a bus is only necessary if the `MutexType` is not `NullMutex`:
```cpp
AZ_Assert(
    (!AZStd::is_polymorphic<typename BusType::InterfaceType>::value
     || AZStd::is_same<typename BusType::MutexType, AZ::NullMutex>::value // <- here
     || !BusIsConnected()),
    "EBus handlers must be disconnected prior to destruction on multi-threaded buses with virtual functions");
```
Otherwise I end up defining a destructor to do what the base class will already do for me. And since I prefer to follow the rule of 6, defining a destructor makes me add all special member functions, even if they are all `= default;`.

I think it is better to inherit that behavior from the base class.